src/lib/ledger_catchup/normal_catchup.ml

```diff
@@ -520,16 +520,18 @@ let verify_transitions_and_build_breadcrumbs ~logger
             ~trust_system ~frontier ~unprocessed_transition_cache transition
         with
         | Error e ->
-            ignore
-              ( List.map acc ~f:Cached.invalidate_with_failure
-                : Mina_block.initial_valid_block Envelope.Incoming.t list ) ;
+            List.iter acc ~f:(fun (node, _vc) ->
+                (* TODO reject callback? *)
```

Comment:
Yes, but it depends on the type of error triggered in `verify_transition`. We probably need to change the error type of that function into something more meaningful in order to handle this.